simtools/util/workflow_description.py

```diff
@@ -0,0 +1,571 @@
+
+        """
+
+        try:
+            return self.workflow_config["DATA_COLUMNS"]
+        except KeyError:
+            self._logger.error(
+                "Missing DATA_COLUMNS entry in workflow configuration")
+            raise
+
+    def product_data_file_name(self, suffix=None):
+        """
+        Return full path and name of product data file
+
+        file name is determined by:
+        a. Toplevel meta ['PRODUCT']['DATA']
+        b. Toplevel meta ['PRODUCT']['ID'] + label
+
+        File name always used CTA:PRODUCT:ID for unique identification
+        (not applied when CONFIGURATION:test is true)
+
+        Returns
+        -------
+        Path
+            data file path and name
+
+        Raises
+        ------
+        KeyError
+            if data file name is not defined in workflow configuration
+            or in product metadata dict
+
+        """
+
+        _directory = self.product_data_directory()
+        try:
+            if self.workflow_config['CONFIGURATION']['test']:
+                _filename = 'TEST'
+            else:
+                _filename = self.workflow_config['ACTIVITY']['ID']
+            if self.workflow_config['PRODUCT']['FILENAME']:
+                _filename += '-' + self.workflow_config['PRODUCT']['FILENAME']
+            else:
+                _filename += '-' + self.workflow_config['ACTIVITY']['NAME']
+        except KeyError:
+            self._logger.error(
+                "Missing CTA:PRODUCT:ID in metadata")
+            raise
+
+        if not suffix:
+            suffix = '.' + self.product_data_file_format(suffix=True)
+
+        return Path(_directory).joinpath(_filename+suffix)
+
+    def product_data_file_format(self, suffix=False):
+        """
+        Return file format for data file
+
+        Parameter
+        ---------
+        suffix: bool
+            return file suffix (if true)
+            return file format (if false)
+
+        Returns
+        -------
+        str
+            file format of data product; default file format is 'ascii.ecsv'
+
+        Raises
+        ------
+        KeyError
+            if relevant fields are not defined in toplevel metadata
+            dictionary
+
+        """
+
+        _file_format = "ascii.ecsv"
+        try:
+            _file_format = self.workflow_config['PRODUCT']['FORMAT']
+        except KeyError:
+            self._logger.info(
+                "Using default file format for model file: ascii.ecsv")
+
+        if suffix and _file_format == "ascii.ecsv":
+            _file_format = "ecsv"
+
+        return _file_format
+
+    def product_data_directory(self):
+        """
+        Return output directory for data products.
+        Create directory if necessary.
+
+        Output directory is determined following this sorted
+        list:
+        1. PRODUCT:DIRECTORY is set (e.g., through workflow file)
+        2. gammasim-tools output location
+
+        Returns
+        -------
+        str
```

Comment:
If I remember correctly the `cfg.get()`, the return type will always be a `path`